azurerm/internal/services/devtestlabs/resource_arm_dev_test_global_shutdown_schedule.go

```diff
@@ -0,0 +1,303 @@
+	defer cancel()
+
+	vmID := d.Get("virtual_machine_id").(string)
+	vm, err := parse.GlobalScheduleVirtualMachineID(vmID)
+	if err != nil {
+		return err
+	}
+
+	vmName := vm.Name
+	resGroup := vm.ResourceGroup
+
+	// Can't find any official documentation on this, but the API returns a 400 for any other name.
+	// The best example I could find is here: https://social.msdn.microsoft.com/Forums/en-US/25a02403-dba9-4bcb-bdcc-1f4afcba5b65/powershell-script-to-autoshutdown-azure-virtual-machine?forum=WAVirtualMachinesforWindows
+	name := "shutdown-computevm-" + vmName
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resGroup, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Schedule %q (Resource Group %q): %s", name, resGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_dev_test_global_shutdown_schedule", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	taskType := "ComputeVmShutdownTask"
+
+	schedule := dtl.Schedule{
+		Location: &location,
+		ScheduleProperties: &dtl.ScheduleProperties{
+			TargetResourceID: &vmID,
+			TaskType:         &taskType,
+		},
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if d.Get("enabled").(bool) {
+		schedule.ScheduleProperties.Status = dtl.EnableStatusEnabled
+	} else {
+		schedule.ScheduleProperties.Status = dtl.EnableStatusDisabled
+	}
+
+	if timeZoneId := d.Get("timezone").(string); timeZoneId != "" {
+		schedule.ScheduleProperties.TimeZoneID = &timeZoneId
+	}
+
+	if v, ok := d.GetOk("daily_recurrence_time"); ok {
+		dailyRecurrence := expandArmDevTestLabGlobalShutdownScheduleRecurrenceDaily(v)
+		schedule.DailyRecurrence = dailyRecurrence
+	}
+
+	if _, ok := d.GetOk("notification_settings"); ok {
+		notificationSettings := expandArmDevTestLabGlobalShutdownScheduleNotificationSettings(d)
+		schedule.NotificationSettings = notificationSettings
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resGroup, name, schedule); err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, name, "")
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Dev Test Global Schedule %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmDevTestLabGlobalShutdownScheduleRead(d, meta)
+}
+
+func resourceArmDevTestLabGlobalShutdownScheduleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DevTestLabs.GlobalLabSchedulesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	schedule, err := parse.GlobalScheduleID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := schedule.ResourceGroup
+	name := schedule.Name
```

Comment:
again we don't need to assign these tolocal alues

Reply:
Removed